Airline_Web_API/Airline_Web_API/Controllers/UsersController.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using Airline_Web_API.Models;
+using Airline_Web_API.DTOs;
+using Airline_Web_API.Services;
+using Airline_Web_API.Helpers;
+
+namespace Airline_Web_API.Controllers
+{
+    [ApiController]
+    [ValidateModelState]
+    [Route("api/users")]
+    public class UsersController : ControllerBase
+    {
+        private readonly AccountService _accountService;
+        private readonly JwtService _jwtService;
+
+        public UsersController(AccountService accountService, JwtService jwtService)
+        {
+            _accountService = accountService;
+            _jwtService = jwtService;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("authenticate")]
+        public async Task<ActionResult> Authenticate([FromBody]AuthenticateModel model)
+        {
+            Response<User> authorizationResult = await _accountService.AuthenticateAsync(model);
+
+            if (authorizationResult.Data == null)
+            {
+                return BadRequest(authorizationResult);
+            }
+
+            string tokenString = _jwtService.GenerateJwtToken(authorizationResult.Data);
+
+            var result = new Response<string>
+            {
+                Message = authorizationResult.Message,
+                Data = tokenString
+            };
+
+            return Ok(result);
+        }
+
+        [AllowAnonymous]
+        [HttpPost("register")]
+        public async Task<ActionResult<Response<string>>> Register([FromBody] RegisterModel model)
+        {
+            Response<string> registrationResult = await _accountService.RegisterAsync(model);
+
+            if (registrationResult.Data == null)
```

Comment:
What if successful request doesn't require to return any data? Maybe it's better to provide some explicit indication of operation success or error?

Reply:
I've added "Success" field which contains true or false depending on the result of action. And then I check the value of this field. If success is false I send bad request.